python/social_bot/envs/grocery_ground.py

```diff
@@ -0,0 +1,336 @@
+            agent_pose = np.array(self._agent.get_pose()).flatten()
+            agent_vel = np.array(self._agent.get_velocities()).flatten()
+            joint_vel = []
+            for joint_id in range(len(self._agent_joints)):
+                joint_name = self._agent_joints[joint_id]
+                joint_state = self._agent.get_joint_state(joint_name)
+                joint_vel.append(joint_state.get_velocities())
+            joint_vel = np.array(joint_vel).flatten()
+            obs_data = np.concatenate(
+                (goal_pose, agent_pose, agent_vel, joint_vel), axis=0)
+            obs_data = np.array(obs_data).reshape(-1)
+        return obs_data
+
+    def step(self, action):
+        """
+        Args:
+            action (dict|int): If with_language, action is a dictionary 
+                    with key "control" and "sentence".
+                    action['control'] is a vector whose dimention is
+                    len(_joint_names). action['sentence'] is a string.
+                    If not with_language, it is an int for the action id.
+        Returns:
+            If with_language, it is a dictionary with key 'data' and 'sentence'
+            If not with_language, it is a numpy.array or image for observation
+        """
+        if self._with_language:
+            sentence = action.get('sentence', None)
+            controls = action['control']
+        else:
+            sentence = ''
+            controls = action
+        controls = dict(zip(self._agent_joints, controls))
+        teacher_action = self._teacher.teach(sentence)
+        self._agent.take_action(controls)
+        self._world.step(10)
+        if self._with_language:
+            obs_data = self._get_observation()
+            obs = OrderedDict(data=obs_data, sentence=teacher_action.sentence)
+        else:
+            obs = self._get_observation()
+        self._steps_in_this_episode += 1
+        self._cum_reward += teacher_action.reward
+        if teacher_action.done:
+            logger.debug("episode ends at cum reward:" + str(self._cum_reward))
+        return obs, teacher_action.reward, teacher_action.done, {}
+
+    def _random_insert_objects(self):
+        for obj_id in range(len(self._object_types)):
+            model_name = self._object_types[obj_id]
+            self._world.insertModelFile('model://' + model_name)
+            logger.debug('model ' + model_name + ' inserted')
+            self._world.step(10)
+            time.sleep(0.1)  # Avoid 'px!=0' error
+
+    def _random_move_objects(self, random_range=10.0):
+        obj_num = len(self._object_types)
+        obj_pos_list = random.sample(self._pos_list, obj_num)
+        for obj_id in range(obj_num):
+            model_name = self._object_types[obj_id]
+            loc = (obj_pos_list[obj_id][0], obj_pos_list[obj_id][1], 0)
+            pose = (np.array(loc), (0, 0, 0))
+            self._world.get_model(model_name).set_pose(pose)
+            self._world.step(10)
```

Comment:
When I run this environment, sometime I still see an object is located at the same position of the agent (0,0,0), which should not happen.

Reply:
The trash can could cause this problem. I can not find the reason. Removed it for now.